subprojects/tooling-api/src/integTest/groovy/org/gradle/integtests/tooling/r36/PluginApplicationBuildProgressCrossVersionSpec.groovy

```diff
@@ -0,0 +1,465 @@
+        given:
+        def events = ProgressEvents.create()
+        settingsFile << """
+            rootProject.name = 'single'
+            apply plugin: ExamplePlugin
+            
+            class ExamplePlugin implements Plugin<Object> {
+                void apply(Object target) { }
+            }
+        """
+        buildFile << ""
+
+        when:
+        withConnection {
+            ProjectConnection connection -> connection.newBuild().addProgressListener(events).run()
+        }
+
+        then:
+        events.assertIsABuild()
+
+        def applySettings = events.operation("Apply settings file '${settingsFile.absolutePath}' to settings '${projectDir.name}'")
+        def examplePlugin = events.operation("Apply plugin 'ExamplePlugin'")
+
+        examplePlugin.parent == applySettings
+    }
+
+    def "generates plugin application events for plugin applied in init script"() {
+        given:
+        def events = ProgressEvents.create()
+        settingsFile << "rootProject.name = 'single'"
+        def initScript = file('init.gradle')
+        buildFile << ""
+        initScript  << """
+            apply plugin: ExamplePlugin
+            
+            class ExamplePlugin implements Plugin<Object> {
+                void apply(Object target) { }
+            }
+        """
+
+        when:
+        withConnection {
+            ProjectConnection connection -> connection.newBuild().addProgressListener(events)
+                .withArguments('--init-script', initScript.toString()).run()
+        }
+
+        then:
+        events.assertIsABuild()
+
+        def applyInitScript = events.operation("Apply initialization script '${initScript.absolutePath}' to build")
+        def examplePlugin = events.operation("Apply plugin 'ExamplePlugin'")
+
+        examplePlugin.parent == applyInitScript
+    }
+
+    def "generates plugin application events for project plugin applied in init script to root project"() {
+        given:
+        def events = ProgressEvents.create()
+        settingsFile << "rootProject.name = 'single'"
+        def initScript = file('init.gradle')
+        buildFile << ""
+        initScript  << """
+            rootProject { 
+                apply plugin: 'java'
+            }
+        """
+
+        when:
+        withConnection {
+            ProjectConnection connection -> connection.newBuild().addProgressListener(events)
+                .withArguments('--init-script', initScript.toString()).run()
+        }
+
+        then:
+        events.assertIsABuild()
+
+        def rootOperation = events.operations[0]
+
+        def java = events.operation("Apply plugin 'org.gradle.java'")
+        def javaBase = events.operation("Apply plugin 'org.gradle.api.plugins.JavaBasePlugin'")
+        def base = events.operation("Apply plugin 'org.gradle.api.plugins.BasePlugin'")
+
+        java.parent == rootOperation.child("Configure project : (initialization scripts)")
```

Comment:
There should be an intermediate build operation here, not a String concatenation of two operations.